entity-registry/src/main/java/com/linkedin/metadata/aspect/plugins/PluginFactory.java

```diff
@@ -105,6 +85,95 @@ public PluginFactory loadPlugins() {
     return this;
   }
 
+  /**
+   * Memory intensive operation because of the size of the jars. Limit packages, classes scanned,
+   * cache results
+   *
+   * @param configs plugin configurations
+   * @return auto-closeable scan result
+   */
+  protected static <T extends PluginSpec> List<T> initPlugins(
+      @Nonnull List<ClassLoader> classLoaders,
+      @Nonnull Class<?> baseClazz,
+      @Nonnull List<String> packageNames,
+      @Nonnull List<AspectPluginConfig> configs) {
+
+    List<String> classNames =
+        configs.stream().map(AspectPluginConfig::getClassName).collect(Collectors.toList());
+
+    if (classNames.isEmpty()) {
+      return List.of();
+    } else {
+      long key =
+          IntStream.concat(
+                  classLoaders.stream().mapToInt(ldr -> ldr.getName().hashCode()),
+                  IntStream.concat(
+                      IntStream.of(baseClazz.getName().hashCode()),
+                      configs.stream().mapToInt(AspectPluginConfig::hashCode)))
+              .sum();
```

Comment:
Interesting, so this is creating a key like
`sum ( classLoaders.hashCode..., aspectPluginConfigClass.hashCode... )`

Is this guaranteed to be unique?

Reply:
This is unique within a given jvm execution. It might leak a bit when/if jars are being constantly replaced since each time that happens we'll be creating orphans. This was primarily added to help with tests where we run multiple concurrent test threads and memory usage was triggering OOM.